### PR TITLE
Simplify running the wireshark demo

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

**Comment by yschimke** [comment]
Output like

```
Capturing on 'Wi-Fi: en0'
Frame 45: 110 bytes on wire (880 bits), 110 bytes captured (880 bits) on interface 0
Ethernet II, Src: Google_fc:86:a2 (28:bd:89:fc:86:a2), Dst: Apple_b3:f3:ee (f0:18:98:b3:f3:ee)
Internet Protocol Version 4, Src: 104.244.42.66, Dst: 192.168.86.193
Transmission Control Protocol, Src Port: 443, Dst Port: 63051, Seq: 3336, Ack: 385, Len: 44
Transport Layer Security
    TLSv1.2 Record Layer: Application Data Protocol: http2
        Content Type: Application Data (23)
        Version: TLS 1.2 (0x0303)
        Length: 39
        Encrypted Application Data: 491e7592a2bc673b8cbc3b8c47bc26e9c89c07dad56bc8b6…
HyperText Transfer Protocol 2
    Stream: SETTINGS, Stream ID: 0, Length 6
        Length: 6
        Type: SETTINGS (4)
        Flags: 0x00
            .... ...0 = ACK: False
            0000 000. = Unused: 0x00
        0... .... .... .... .... .... .... .... = Reserved: 0x0
        .000 0000 0000 0000 0000 0000 0000 0000 = Stream Identifier: 0
        Settings - Initial Windows size : 65536
            Settings Identifier: Initial Windows size (4)
            Initial Windows Size: 65536
```

File: samples/guide/src/main/java/okhttp3/recipes/kt/WiresharkExample.kt [hunk 1]
```diff
@@ -47,124 +51,177 @@ import okio.ByteString.Companion.toByteString
...
-        if (message == "Produced ClientHello handshake message") {
-          random = readClientRandom(parameter)
+          Thread.sleep(10000)
         }
       }
+      CommandLine -> {
+        return ProcessBuilder(
+            "tshark", "-l", "-V", "-o", "tls.keylog_file:$logFile", "-Y", "http2", "-O", "http2,tls")
+            .redirectInput(File("/dev/null"))
+            .redirectOutput(Redirect.INHERIT)
```
Comment (yschimke) [inline samples/guide/src/main/java/okhttp3/recipes/kt/WiresharkExample.kt:93]: Could maybe be STDERR instead?

File: samples/guide/src/main/java/okhttp3/recipes/kt/WiresharkExample.kt [hunk 2]
```diff
@@ -186,37 +243,30 @@ class WireSharkKeyLoggerListener(
...
-class WiresharkExample(private val tlsVersions: List<TlsVersion>) {
-  val connectionSpec =
+class WiresharkExample(tlsVersions: List<TlsVersion>, private val launch: Launch? = null) {
+  private val connectionSpec =
     ConnectionSpec.Builder(ConnectionSpec.RESTRICTED_TLS)
         .tlsVersions(*tlsVersions.toTypedArray())
         .build()
 
+  private val eventListenerFactory = WireSharkListenerFactory(
+      logFile = File("/tmp/key.log"), tlsVersions = tlsVersions, launch = launch)
```
Comment (yschimke) [inline samples/guide/src/main/java/okhttp3/recipes/kt/WiresharkExample.kt:260]: Passing tlsVersions here is a bit clumsy... But makes the copy and paste keep the warning messages on.  I could react to the negotiated connection instead.

File: samples/guide/src/main/java/okhttp3/recipes/kt/WiresharkExample.kt [hunk 3]
```diff
@@ -47,124 +51,177 @@ import okio.ByteString.Companion.toByteString
...
+          println("TLSv1.3 requires an external command run before first traffic is sent")
+          println("Follow instructions at https://github.com/neykov/extract-tls-secrets for TLSv1.3")
+          println("Pid: ${ProcessHandle.current().pid()}")
 
-        if (message == "Produced ClientHello handshake message") {
-          random = readClientRandom(parameter)
+          Thread.sleep(10000)
         }
       }
+      CommandLine -> {
```
Comment (swankjesse) [inline samples/guide/src/main/java/okhttp3/recipes/kt/WiresharkExample.kt:89]: Sweet

**Comment by swankjesse** [comment]
Nice!

**Pull request merged**